Makefile

```diff
@@ -19,25 +19,31 @@
 
 RUST_LOG = debug
 
+RUST_NIGHTLY_CHANNEL = nightly-2024-06-10
```

Comment:
Hi, as I proposed in https://github.com/apache/iceberg-rust/issues/440, we can write `nightly-2024-06-10` in `rust-toolchain.toml`.

Reply:
> Do you mean to use nightly to build?

Yes. We can add extra build task with `cargo +stable build` to make sure it work in stable channel.

Reply:
Hi, my current plan is we have three rust version that need to test:

- Development: the nightly version we picked to develop. The commit could be stable or MSRV.
- MSRV: the MSRV we promised for downstream users.
- Stable: the latest stable version of rust.